norddrop/src/device/types.rs

```diff
@@ -53,15 +53,39 @@ pub struct ProgressEvent {
     transfered: u64,
 }
 
+#[derive(Serialize)]
+pub struct Status {
+    status: u32,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    os_error_code: Option<i32>,
```

Comment:
OS errours should be unique along the whole OS, meaning 13(permissions) should only mean that unless we "eat" it in some wrapper and provide our own error. I assumed from the code that we clearly know when we have OS error instead of Rust one?

Reply:
So you simply want to drop the `status` in case we know the OS code?

Reply:
Yes. I think it only simplifies things and allows for easier understanding of the code and as a result - the API

Reply:
OK, so the decision here is to leave it as it is for now